knowledge-base/grid-drag-drop-rows-hierarchy.md

```diff
@@ -0,0 +1,261 @@
+---
+title: Drag and Drop Grid Rows in Hierarchy
+description: Learn how to implement and setup drag-and-drop in hierarchical Blazor Grid scenarios. Enable or disable dragging and dropping in specific Grid instances in the hierarchy.
+type: how-to
+page_title: How to Drag and Drop Grid Rows in Hierarchy
+slug: grid-kb-drag-drop-rows-hierarchy
+tags: blazor, grid, hierarchy, drag and drop
+ticketid: 1675157
+res_type: kb
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>
+                Grid for Blazor
+            </td>
+        </tr>
+    </tbody>
+</table>
+
+## Description
+
+This KB answers the following questions:
+
+* How to configure drag-and-drop in a hierarchy of Telerik Grids for Blazor?
+* How to drag detail row from a child Grid and drop them on the master (parent) Grid?
+* How to prevent users from dragging and dropping master Grid rows to detail (child) Grids?
+
+## Solution
+
+1. Configure the [Grid hierarchy](slug://components/grid/features/hierarchy) with a `<DetailTemplate>`.
+1. [Set `RowDraggable="true"` and handle the `OnRowDrop` event](slug://grid-drag-drop-overview) for the master or detail Grids.
+1. [Set unique `@ref` attributes to all detail Grid instances](slug://common-kb-dynamic-refs).
```

Comment:
Wouldn't the `@ref` and `@key` steps happen in the same step for each Grid? Or is it more likely for people to go over the code, set `@ref` for all Grids and then go back and set `@key` for all Grids?